spec/unit/type/service_spec.rb

```diff
@@ -1,7 +1,22 @@
 #! /usr/bin/env ruby
 require 'spec_helper'
 
-describe Puppet::Type.type(:service) do
+def safely_load_service_type
+  before(:all) do
+    # We have a :confine block that calls execute in our upstart provider, which fails
+    # on jruby. Thus, we stub it out here since we don't care to do any assertions on it.
+    # This is only an issue if you're running these unit tests on a platform where upstart
+    # is a default provider, like Ubuntu trusty.
+    Puppet::Util::ExecutionStub.set { |_, _| '' }
+    Puppet::Type.type(:service)
+  end
+end
```

Comment:
Our spec_helper ends up calling `ExecutionStub.reset` at the end of each test https://github.com/puppetlabs/puppet/blob/master/lib/puppet/test/test_helper.rb#L163. So I think this needs to be `before(:each)`. Might be better to use `around` so it's explicit.

```ruby
around(:each) do |example|
  Puppet::Util::ExecutionStub.set { |_, _| '' }
  example.run
ensure
  Puppet::Util::ExecutionStub.reset
end
```

Reply:
Ah. I wanted to use `Puppet::Util::Execution.stubs(:execute)` but couldn't do it around a `before(:all)` block (it's only defined in `it` groups and `before(:each)` blocks, I think) so that's why I switched it over to the ExecutionStub class. Seeing as `before(:all)` doesn't work either per your comment, I figured it'd be easier to use `stubs(:execute)` instead.

I'm happy to change it to your suggestion too.